kivy/uix/behaviors/touchripple.py

```diff
@@ -0,0 +1,262 @@
+class TouchRippleBehavior(object):
+    '''Touch ripple behavior.
+
+    Supposed to be used as mixin on widget classes.
+
+    Ripple behavior does not trigger automatically, concrete implementation
+    needs to call :func:`ripple_show` respective :func:`ripple_fade` manually.
+
+    Example
+    -------
+
+    Here we create a Label which renders the touch ripple animation on
+    interaction::
+
+        class RippleLabel(TouchRippleBehavior, Label):
+
+            def __init__(self, **kwargs):
+                super(RippleLabel, self).__init__(**kwargs)
+
+            def on_touch_down(self, touch):
+                collide_point = self.collide_point(touch.x, touch.y)
+                if collide_point:
+                    touch.grab(self)
+                    self.ripple_show(touch)
+                    return True
+                return False
+
+            def on_touch_up(self, touch):
+                if touch.grab_current is self:
+                    touch.ungrab(self)
+                    self.ripple_fade()
+                    return True
+                return False
+    '''
+
+    ripple_rad_default = NumericProperty(10)
+    '''Default radius the animation starts from.
+
+    :attr:`ripple_rad_default` is a :class:`~kivy.properties.NumericProperty`
+    and defaults to `10`.
+    '''
+
+    ripple_duration_in = NumericProperty(.5)
+    '''Animation duration taken to show the overlay.
+
+    :attr:`ripple_duration_in` is a :class:`~kivy.properties.NumericProperty`
+    and defaults to `0.5`.
+    '''
+
+    ripple_duration_out = NumericProperty(.2)
+    '''Animation duration taken to fade the overlay.
+
+    :attr:`ripple_duration_out` is a :class:`~kivy.properties.NumericProperty`
+    and defaults to `0.2`.
+    '''
+
+    ripple_fade_from_alpha = NumericProperty(.5)
+    '''Alpha channel for ripple color the animation starts with.
+
+    :attr:`ripple_fade_from_alpha` is a
+    :class:`~kivy.properties.NumericProperty` and defaults to `0.5`.
+    '''
+
+    ripple_fade_to_alpha = NumericProperty(.8)
+    '''Alpha channel for ripple color the animation targets to.
+
+    :attr:`ripple_fade_to_alpha` is a :class:`~kivy.properties.NumericProperty`
+    and defaults to `0.8`.
+    '''
+
+    ripple_scale = NumericProperty(2.)
+    '''Max scale of the animation overlay calculated from max(width/height) of
+    the decorated widget.
+
+    :attr:`ripple_scale` is a :class:`~kivy.properties.NumericProperty`
+    and defaults to `2.0`.
+    '''
+
+    ripple_func_in = StringProperty('in_cubic')
+    '''Animation callback for showing the overlay.
+
+    :attr:`ripple_func_in` is a :class:`~kivy.properties.StringProperty`
+    and defaults to `in_cubic`.
+    '''
+
+    ripple_func_out = StringProperty('out_quad')
+    '''Animation callback for hiding the overlay.
+
+    :attr:`ripple_func_out` is a :class:`~kivy.properties.StringProperty`
+    and defaults to `out_quad`.
+    '''
+
+    ripple_rad = NumericProperty(ripple_rad_default)
```

Comment:
Does this actually run for you?

ripple_rad_default here is a NumericProperty so its value cannot be accessed like this